WordPress/src/main/java/org/wordpress/android/ui/stories/prefs/StoriesPrefs.kt

```diff
@@ -32,13 +35,60 @@ class StoriesPrefs @Inject constructor(
                 KEY_PREFIX_LOCAL_MEDIA_ID + mediaId.value.toString()
     }
 
+    private fun buildSlideKey(siteId: Long, tempId: TempId): String {
+        return KEY_PREFIX_STORIES_SLIDE_ID + siteId.toString() + "-" +
+                KEY_PREFIX_TEMP_MEDIA_ID + tempId.id
+    }
+
+    @SuppressLint("ApplySharedPref")
+    @Synchronized
+    fun getNewIncrementalTempId(): Long {
+        var currentIncrementalId = getIncrementalTempId()
+        currentIncrementalId++
+        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
+        editor.putLong(KEY_STORIES_SLIDE_INCREMENTAL_ID, currentIncrementalId)
+        editor.commit()
+        return currentIncrementalId
+    }
+
+    fun getIncrementalTempId(): Long {
+        return PreferenceManager.getDefaultSharedPreferences(context).getLong(
+                        KEY_STORIES_SLIDE_INCREMENTAL_ID,
+                        0
+                )
+    }
+
     fun checkSlideIdExists(siteId: Long, mediaId: RemoteId): Boolean {
         val slideIdKey = buildSlideKey(siteId, mediaId)
         return PreferenceManager.getDefaultSharedPreferences(context).contains(slideIdKey)
     }
 
+    fun checkSlideIdExists(siteId: Long, tempId: TempId): Boolean {
+        val slideIdKey = buildSlideKey(siteId, tempId)
+        return PreferenceManager.getDefaultSharedPreferences(context).contains(slideIdKey)
+    }
+
+    fun checkSlideIdExists(siteId: Long, localId: LocalId): Boolean {
+        val slideIdKey = buildSlideKey(siteId, localId)
+        return PreferenceManager.getDefaultSharedPreferences(context).contains(slideIdKey)
+    }
+
     fun checkSlideOriginalBackgroundMediaExists(siteId: Long, mediaId: RemoteId): Boolean {
         val storyFrameItem: StoryFrameItem? = getSlideWithRemoteId(siteId, mediaId)
+        return checkSlideOriginalBackgroundMediaExists(storyFrameItem)
+    }
+
+    fun checkSlideOriginalBackgroundMediaExists(siteId: Long, tempId: TempId): Boolean {
+        val storyFrameItem: StoryFrameItem? = getSlideWithTempId(siteId, tempId)
+        return checkSlideOriginalBackgroundMediaExists(storyFrameItem)
+    }
+
+    fun checkSlideOriginalBackgroundMediaExists(siteId: Long, localId: LocalId): Boolean {
+        val storyFrameItem: StoryFrameItem? = getSlideWithLocalId(siteId, localId)
+        return checkSlideOriginalBackgroundMediaExists(storyFrameItem)
+    }
```

Comment:
Nitpick comments here :) We could make these private and also when I utilize a sealed class for a type I normally make the variable name more descriptive since the type gives enough context. so I would go with `mediaId: LocalId` so it's easy to see that we are representing the local media id here because `localId` on it's own isn't as clear. Same thought process for the `checkSlideIdExists` above. Again, this is a Nitpick 😅

Reply:
haha okay. You always make a good argument so I see what you are saying as well but yes, lets your changes look good and I still get the same point based on the type in use 😄